assets/js/admin.js

```diff
@@ -198,15 +198,20 @@ function populateTemplate( item, dialog, action ) {
 					</form>
 					<promote-job-template>
 						<div slot="buttons" class="promote-buttons-group">
-								<a class="promote-button button button-primary" target="_blank" href="${ this.getAttribute( 'data-href' ) }">${ job_manager_admin_params.job_listing_promote_strings.promote_job }</a>
-								<a class="promote-button button button-secondary" target="_blank" href="#">${ job_manager_admin_params.job_listing_promote_strings.learn_more }</a>
+							<a id="wpjm-promote-button" class="promote-button button button-primary" target="_blank" rel="noopener noreferrer" href="${ this.getAttribute( 'data-href' ) }">${ job_manager_admin_params.job_listing_promote_strings.promote_job }</a>
+							<a class="promote-button button button-secondary" target="_blank" rel="noopener noreferrer" href="#">${ job_manager_admin_params.job_listing_promote_strings.learn_more }</a>
 						</div>
 					<promote-job-template>`;
 				}
+
 				if ( 'deactivate' === action ) {
 					let deactivateButton = dialog.querySelector( '.deactivate-promotion' );
 					deactivateButton.setAttribute( 'href', this.getAttribute( 'data-href' ) );
 				}
+
+				document.querySelector( '#wpjm-promote-button' ).addEventListener( 'click', function() {
```

Comment:
This causes an error if we click to deactivate a button, because then `#wpjm-promote-button` isn't defined. Maybe we should have this only if `'promote' === action`?

Also, wouldn't it be better to have the event listener on `dialog` directly, maybe? Or maybe just use `dialog.querySelector`? I'm thinking about maybe reducing the scope of the event listener.

Reply:
Good catch! Updated here: https://github.com/Automattic/WP-Job-Manager/pull/2508/commits/4337c533033ae0f9334d7e02fb69dab5be9ab999